eras/dijkstra/impl/src/Cardano/Ledger/Dijkstra/TxBody.hs

```diff
@@ -380,6 +388,13 @@ instance
         23
           | STopTx <- sTxLevel ->
               fieldAA (subTransactionsDijkstraTxBodyRawL .~) (D decodeSubTransactions)
+        24
+          | SSubTx <- sTxLevel ->
+              fieldAGuarded
+                (emptyFailure "RequiredTopLevelGuards" "non-empty")
+                Map.null
+                (requiredTopLevelGuardsDijkstraTxBodyRawL .~)
+                (D (decodeMapByKey decCBOR (const decCBOR)))
```

Comment:
For some reason this doesn't seem to fail on duplicates, even though looking at the implementation it should!  I will investigate in a few hours